src/other-examples/react-window/index.js

```diff
@@ -0,0 +1,180 @@
+import React, {PureComponent} from 'react'
+import {FixedSizeList as List} from 'react-window'
+import starWarsNames from 'starwars-names'
+import Downshift from 'downshift'
+import {
+  Label,
+  Menu,
+  ControllerButton,
+  Input,
+  Item,
+  ArrowIcon,
+  XIcon,
+  css,
+  getItems,
+  itemToString,
+} from '../../shared'
+
+class ItemRenderer extends PureComponent {
+  render() {
+    const {
+      items,
+      getItemProps,
+      highlightedIndex,
+      selectedItem,
+    } = this.props.data
+    const item = items[this.props.index]
+    return (
+      <Item
+        {...getItemProps({
+          style: this.props.style,
+          item,
+          index: this.props.index,
+          isActive: highlightedIndex === this.props.index,
+          isSelected: selectedItem === item,
+        })}
+      >
+        {itemToString(item)}
+      </Item>
+    )
+  }
+}
+
+class ExampleDownshift extends React.Component {
+  listRef = React.createRef()
+
+  scrollToItem = highlightedIndex => {
+    if (this.listRef.current !== null) {
+      this.listRef.current.scrollToItem(highlightedIndex)
+    }
+  }
+
+  stateReducer = (state, changes) => {
+    // this scrolls the react-window list to the highlightedIndex
+    switch (changes.type) {
+      case Downshift.stateChangeTypes.keyDownArrowUp:
+        this.scrollToItem(changes.highlightedIndex)
```

Comment:
Doing side-effects like this is discouraged. I recommend doing this in an `onStateChange` callback. Other than that this looks great!

Reply:
Hi Kent,

thank's you for your input :). I updated the code to use onStateChange and also removed some things I have missed while preparing the pull request.